File: client-app/src/app/models/post.ts
export interface IPostsEnvelope {
  posts: IPost[];
  postCount: number;
}

export interface IPost {
  id: string;
  title: string;
  content: string;
  description: string;
  thumbnail: IPhoto;
  category: string;
  createdAt: string;
  updatedAt: string;
  host: IHost;
}

export class PostFormValues {
  id: string = "";
  title: string = "";
  content: string = "";
  description: string = "";
  thumbnail: IPhoto = { id: "", url: "" };
  category: string = "";
  constructor(init?: PostFormValues) {
    Object.assign(this, init);
  }
}

export interface IPhoto {
  id: string;
  url: string;
}

export interface IHost {
  username: string;
  displayName: string;
  image: string;
}

File: Domain/Post.cs
using System;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Content { get; set; }
        public string Category { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public string AppUserId { get; set; }
        public virtual Photo Thumbnail { get; set; }
        public virtual AppUser AppUser { get; set; }
    }
}